src/libs/cluster-utils.js

```diff
@@ -51,13 +51,9 @@ export const runtimeConfigCost = config => {
   ])
 }
 
-export const persistentDiskCostMonthly = config => {
-  return config.size * monthlyStoragePrice
-}
-
-export const persistentDiskCost = config => {
-  return config.size * storagePrice
-}
+export const [persistentDiskCost, persistentDiskCostMonthly] = _.map(price => ({ size, status }) => {
+  return _.includes(status, ['Deleting', 'Failed']) ? 0.0 : size * price
+}, [storagePrice, monthlyStoragePrice])
```

Comment:
It's clear to me in the context of this PR (since I can see the before and after of the refactor), however, it does take a bit of mental energy for me to read this, and I know I can't immediately forget my current knowledge of the intention to create 2 functions with the same implementation. Reading code that generates code is a difficult and infrequently used skill.

Maybe extracting the higher order function might help? Something like:

```
const generateDiskCostFunction = price => ({ size, status }) => {
  return _.includes(status, ['Deleting', 'Failed']) ? 0.0 : size * price
}
export const persistentDiskCost = generateDiskCostFunction(storagePrice)
export const persistentDiskCostMonthly = generateDiskCostFunction(monthlyStoragePrice)
```

Giving the function a name allows reuse without having to add the complexity of calling `_.map` and destructuring the result.

Reply:
Thanks, that was the other way I was considering, but I was having trouble coming up with a name.

Reply:
I do think this is easier to read, seeing both ways. So +1 on Brian's comment.

Reply:
Yeah, naming is tough. I'm not committed to the name I suggested. I did choose the `generate` prefix because I don't think we use it extensively in any other context, and maybe we can have a convention of using it as a clue that we're in a code generation mindset.